validators.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	v1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	log "github.com/sirupsen/logrus"
+)
+
+func PendingJobs(maxAge float64, kubeClient kubernetes.Interface) func(v1.Job) bool {
+	return func(job v1.Job) bool {
+		podList, err := kubeClient.CoreV1().Pods("").List(metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("job-name=%s", job.ObjectMeta.Name),
+		})
+
+		if err != nil {
+			log.Fatal(err.Error())
+		}
+
+		for _, pod := range podList.Items {
+			if pod.Status.Phase == "Pending" {
+				if pod.ObjectMeta.CreationTimestamp.Time.IsZero() {
+					continue
+				}
+				age := time.Since(pod.ObjectMeta.CreationTimestamp.Time)
+				if age.Minutes() >= maxAge {
+					return true
+				}
+			}
+		}
+
+		return false
+	}
+}
+
+func ExpiredJobs(maxAge float64, annotation string) func(v1.Job) bool {
+	expirationAnnotationName := fmt.Sprintf("%s/expiration", annotation)
+
+	return func(job v1.Job) bool {
+		if job.Status.CompletionTime == nil {
+			return false
+		}
+
+		age := time.Since(job.Status.CompletionTime.Time)
+		expirationAnnotation := job.ObjectMeta.Annotations[expirationAnnotationName]
+
+		maxAgeOverride, err := strconv.ParseFloat(expirationAnnotation, 64)
+		if err == nil {
+			log.Debugf(
+				"Expiration override for (%s:%s) with annotation (%s) of (%s)",
+				job.ObjectMeta.Namespace,
+				job.ObjectMeta.Name,
+				expirationAnnotationName,
+				expirationAnnotation,
+			)
+			if age.Minutes() >= maxAgeOverride {
```

Comment:
For these last two conditions, we can be a little bit more clever. Instead of.
```Python
if thing > 100:
  return true
return false
```
we can do
```Python
return thing > 100
```

Reply:
Addressed in https://github.com/theMagicalKarp/kube-janitor/pull/6/commits/d7db5c0f2cc7a69d8332beff36489d9456fa6c80